ifermi/plot.py

```diff
@@ -346,7 +353,10 @@ def _get_plot_data(
 
         isosurfaces = []
         if not hide_surface:
-            isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            if plot_index is None:
+                isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            else:
+                isosurfaces = [self.fermi_surface.all_vertices_faces(spins=spin)[int(i)] for i in plot_index]
```

Comment:
This is broken if the system is spin polarised. In that case, the `all_vertices_faces` will be a list of `[spin_1_band_1, spin_1_band_2, .... spin_2_band_1, spin_2_band_2, ....]`. So lets say you want to select band 5 in the spin up and spin down channel, you can't do that. Or alternatively, if you want to select band 5 off the spin down channel only.

There is a second issue, which is that you select the bands for the Fermi surface faces, but you don't select the bands for the Fermi surface properties (which happens several lines below this). This means there will be a mismatch between the Fermi surfaces and their properties.

Thinking about this, I think the cleanest way would be to add a `band_index` option to `FermiSurface.all_vertices_faces`, `FermiSurface.all_properties`, `FermiSlice.all_lines`, `FermiSlice.all_properties`. This could then be used cleanly in the `_get_plot_data` functions.

The `band_index` argument should accept:
1. A single integer, which will select that band index in both spin channels (if both spin channels are present).
2. A list of integers, which will select that set of bands from both spin channels (if both a present).
3. A dictionary of {Spin: int}, which will specifies both the spin and the band.
4. A dictionary of {Spin: list[int]} which is like option 3 but allows you to specify multiple bands for a spin channel.

E.g., all of these should be work:
```python
band_index = 1
band_index = [2, 3, 4]
band_index = {Spin.up: 1}
band_index = {Spin.down: 5}
band_index = {Spin.up: 1, Spin.down: 5}
band_index = {Spin.up: [2, 3, 4]}
band_index = {Spin.up: [2, 3, 4], Spin.down: 5}
etc
```

I realise this is a bit of extra work, but I think it will make this the most robust method.